ui/src/cohortUpdaterContext.test.tsx

```diff
@@ -0,0 +1,38 @@
+import { Cohort, GroupKind } from "cohort";
+import { ConceptCriteria } from "criteria/concept";
+import { CohortUpdater } from "./cohortUpdaterContext";
+
+test("updates", () => {
+  var cohort = new Cohort("name", "underlay", "entity");
+  const setCohort = (c: Cohort) => {
+    cohort = c;
+  };
+
+  expect(cohort.groups.length).toBe(0);
+
+  var prevCohort = cohort;
+  new CohortUpdater(cohort, setCohort).update((c: Cohort) => {
+    c.addGroupAndCriteria(
+      GroupKind.Included,
+      new ConceptCriteria("test-name", "test-filter")
+    );
+  });
+
+  expect(cohort).not.toBe(prevCohort);
+  expect(cohort.groups.length).toBe(1);
+  expect(cohort.groups[0].criteria.length).toBe(1);
+  expect(cohort.groups[0].criteria[0].name).toBe("test-name");
+
+  const group = cohort.groups[0];
+  new CohortUpdater(cohort, setCohort).updateCriteria(
+    group.id,
+    group.criteria[0].id,
+    (criteria: ConceptCriteria) => {
+      criteria.name = "new-test-name";
+    }
+  );
+
+  expect(cohort.groups.length).toBe(1);
+  expect(cohort.groups[0].criteria.length).toBe(1);
```

Comment:
just a suggestion, but there is probably a more jest-y way of doing this...

expect(cohort.groups[0].criteria.length).toBe([expect.objectContaining[{name: "new-test-name"}])

or even more top level.